=== modified file 'sources/mainwindow.cpp'
--- sources/mainwindow.cpp	2020-01-03 10:49:05 +0000
+++ sources/mainwindow.cpp	2021-01-19 17:20:20 +0000
@@ -3312,11 +3312,11 @@
     dock->setObjectName(objectName);
     dock->setFeatures(QDockWidget::DockWidgetClosable | QDockWidget::DockWidgetFloatable | QDockWidget::DockWidgetMovable);
 
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
 
     dock->setWindowTitle(dock->windowTitle().remove(QLatin1Char('&')));
 
-#endif // Q_OS_WIN
+#endif // defined(Q_OS_WIN) || defined(Q_OS_MAC)
 
     addDockWidget(Qt::LeftDockWidgetArea, dock);
 
@@ -3338,11 +3338,11 @@
     m_searchDock->setObjectName(QLatin1String("searchDock"));
     m_searchDock->setFeatures(QDockWidget::DockWidgetFloatable | QDockWidget::DockWidgetMovable | QDockWidget::DockWidgetVerticalTitleBar);
 
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
 
     m_searchDock->setWindowTitle(m_searchDock->windowTitle().remove(QLatin1Char('&')));
 
-#endif // Q_OS_WIN
+#endif // defined(Q_OS_WIN) || defined(Q_OS_MAC)
 
     addDockWidget(Qt::BottomDockWidgetArea, m_searchDock);
 

