=== modified file 'sources/pageitem.cpp'
--- sources/pageitem.cpp	2017-04-19 21:01:25 +0000
+++ sources/pageitem.cpp	2019-12-30 20:57:31 +0000
@@ -575,7 +577,22 @@
     {
         if(m_boundingRect.contains(event->pos()))
         {
-            m_rubberBand.setBottomRight(event->pos());
+            // At least on macOS it is possible to receive {x,y,-1,-1} -> {x,y,0,0} rect at a start.
+            // setBottomRight will effectively make m_rubberBand null and will not selection to start.
+            // As a workaround we make sure the selection is at least 1x1.
+            QPointF newPos = event->pos();
+
+            if(newPos.x() == m_rubberBand.x())
+            {
+                newPos.setX(m_rubberBand.x() + 1);
+            }
+
+            if(newPos.y() == m_rubberBand.y())
+            {
+                newPos.setY(m_rubberBand.y() + 1);
+            }
+
+            m_rubberBand.setBottomRight(newPos);
 
             update();
 
