=== modified file 'sources/documentview.cpp'
--- sources/documentview.cpp	2021-01-17 10:13:17 +0000
+++ sources/documentview.cpp	2021-01-19 17:27:14 +0000
@@ -981,7 +981,16 @@
         title = m_fileInfo.completeBaseName();
     }
 
+#ifdef Q_OS_MAC
+    // On macOS latest Qt versions have a bug of automatically detecting tab entries
+    // as menu entries with roles, causing opening files with these names result in
+    // "Preferences" window breaking. The relevant code can be found here:
+    // https://github.com/qt/qtbase/blob/1869615fc959c70a334e666ebf95ff595a3d6e67/src/plugins/platforms/cocoa/qcocoamenuitem.mm#L228-L234
+    // Reported to qpdfview: https://bugs.launchpad.net/qpdfview/+bug/1912274
+    return QString(u8"\u200B") + title;
+#else
     return title;
+#endif
 }
 
 QStringList DocumentView::openFilter()

